src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/EmbeddedLanguages/VirtualChars/AbstractVirtualCharService.cs

```diff
@@ -166,7 +166,16 @@ private void CheckInvariants(SyntaxToken token, VirtualCharSequence result)
 
                     result.Add(VirtualChar.Create(new Rune(tokenText[index]), span));
                     index += result[^1].Span.Length;
-                    continue;
+                }
+                else
+                {
+                    var (virtualChar, consumedCharacters) = VirtualChar.CreateNextInString(
+                        tokenText,
+                        index,
+                        offset);
+
+                    result.Add(virtualChar);
+                    index += consumedCharacters;
                 }
 
                 index += ConvertTextAtIndexToRune(tokenText, index, result, offset);
```

Comment:
This seems to already be handled by the `else` clause that you added. By having it here this code is executing twice and some of the text is not covered by the earlier quote and brace checks.

```suggestion
```

Reply:
Not entirely sure I follow. If we move the characters by a certain amount the index will always need to be added to. `CreateNextInString` will consume some number of characters to produce a `virtualChar`. That's separate from `ConvertTextAtIndexToRune`